vaadin-accordion-flow-parent/vaadin-accordion-flow/src/main/java/com/vaadin/flow/component/accordion/AccordionPanel.java

```diff
@@ -32,6 +33,19 @@
 @JsModule("@vaadin/accordion/src/vaadin-accordion-panel.js")
 public class AccordionPanel extends Details {
 
+    private Component summary;
+    private Component summaryContainer;
+
+    /**
+     * Server-side component for the {@code <vaadin-accordion-heading>} element.
+     */
+    @Tag("vaadin-accordion-heading")
+    static class AccordionHeading extends Component {
```

Comment:
Previously, `vaadin-details` and `vaadin-accordion` were accepting any element with `slot="summary"`.
This has changed now so both have individual web components with different ARIA semantics:

- `vaadin-accordion-heading` is an element with `role="heading"` and `<button>` in Shadow DOM,
- `vaadin-details-summary` is an element with `role="button"` itself (no extra element involved).

Therefore I had to updat methods related to setting summary text / component accordingly.